src/lib/checkVersion.ts

```diff
@@ -93,7 +79,48 @@ export class CheckVersion {
     }
   }
 
-  async compareVersions(packageJson: string, packageLock: string) {
+  getVersion(manager: ManagerType, location: string) {
+    if (manager === 'cargo') return this.handleCargo(location)
+    if (manager === 'npm') return this.handlePackageJson(location)
+
+    throw new Error(`unknown manager type - [${manager}]`)
+  }
+
+  async handleCargo(location: string) {
+    const cargo: Cargo = new Cargo(this.fs)
+    const result = await cargo.scan(location)
+
+    if (result) return result.version
+    
+    return result 
+  }
+
+  async handlePackageJson(location: string, packageJson: {version?: string} | null = null, packageLockJson: any = null) {
```

Comment:
It would be good to refactor this at some point into an `Npm` class at some point to mirror the structure of `Cargo`. Maybe add a ticket to the backlog once this merges?

Reply:
https://digicatapult.atlassian.net/browse/L3-198 done.